pkg/docker/config/config.go

```diff
@@ -0,0 +1,279 @@
+	ErrNotLoggedIn = errors.Errorf("not logged in")
+)
+
+// SetAuthentication stores the username and password in the auth.json file
+func SetAuthentication(registry, username, password string) error {
+	registry = dockerHostNameToRegistry(registry)
+	return modifyJSON(func(auths *dockerConfigFile) (bool, error) {
+		if ch, exists := auths.CredHelpers[registry]; exists {
+			if err := setAuthToCredHelper(ch, registry, username, password); err != nil {
+				return false, err
+			}
+			return true, nil
+		}
+
+		creds := base64.StdEncoding.EncodeToString([]byte(username + ":" + password))
+		newCreds := dockerAuthConfig{Auth: creds}
+		auths.AuthConfigs[registry] = newCreds
+		return true, nil
+	})
+}
+
+// GetAuthentication returns the registry credentials stored in
+// either auth.json file or .docker/config.json
+func GetAuthentication(registry string) (string, string, error) {
+	registry = dockerHostNameToRegistry(registry)
+	// Check auth.json file first
+	path, err := getPathToAuth()
+	if err != nil {
+		return "", "", errors.Wrapf(err, "error getting path to auth file")
+	}
+	username, password, err := findAuthentication(registry, path)
+	if err != nil {
+		return "", "", err
+	}
+	if username != "" && password != "" {
+		return username, password, nil
+	}
+
+	// check ~/.docker/config.json as secondary option
+	path = filepath.Join(homedir.Get(), dockerCfg, dockerCfgFileName)
+	username, password, err = findAuthentication(registry, path)
+	if err != nil {
+		return "", "", err
+	}
+	return username, password, nil
+}
+
+// RemoveAuthentication deletes the credentials stored in auth.json
+func RemoveAuthentication(registry string) error {
+	registry = dockerHostNameToRegistry(registry)
+	return modifyJSON(func(auths *dockerConfigFile) (bool, error) {
+		// First try cred helpers.
+		if ch, exists := auths.CredHelpers[registry]; exists {
+			if err := deleteAuthFromCredHelper(ch, registry); err != nil {
+				return false, err
+			}
+			return true, nil
+		}
+
+		if _, ok := auths.AuthConfigs[registry]; ok {
+			delete(auths.AuthConfigs, registry)
+		} else if _, ok := auths.AuthConfigs[normalizeRegistry(registry)]; ok {
+			delete(auths.AuthConfigs, normalizeRegistry(registry))
+		} else {
+			return false, ErrNotLoggedIn
+		}
+		return true, nil
+	})
+}
+
+// dockerHostNameToRegistry converts the docker hostname to the docker registry
+func dockerHostNameToRegistry(registry string) string {
+	if registry == dockerHostname {
+		return dockerRegistry
+	}
+	return registry
+}
+
+// getPath gets the path of the auth.json file since the user can set their own path
+// by setting CRIO_AUTH_PATH
+// If CRIO_AUTH_PATH is not set, the default path is ~/.config/containers/auth.json
+func getPathToAuth() (string, error) {
+	if os.Getenv("CRIO_AUTH_PATH") != "" {
+		return filepath.Join(os.Getenv("CRIO_AUTH_PATH"), authCfgFileName), nil
+	}
+	dir := filepath.Join(homedir.Get(), authCfg)
+	return filepath.Join(dir, authCfgFileName), nil
```

Comment:
These two can be collapsed into a single `filepath.Join` call.

Reply:
fixed.